packages/editor/src/components/block-settings-menu/reusable-block-convert-button.js

```diff
@@ -49,25 +49,33 @@ export function ReusableBlockConvertButton( {
 
 export default compose( [
 	withSelect( ( select, { clientIds } ) => {
-		const { getBlock, getReusableBlock } = select( 'core/editor' );
+		const { getBlock, canInsertBlockType, getReusableBlock } = select( 'core/editor' );
 		const { getFallbackBlockName } = select( 'core/blocks' );
 
 		const blocks = map( clientIds, ( clientId ) => getBlock( clientId ) );
+		if ( some( blocks, ( block ) => ! block ) ) {
+			return { isVisible: false };
+		}
 
 		// Hide 'Add to Reusable Blocks' on Classic blocks. Showing it causes a
 		// confusing UX, because of its similarity to the 'Convert to Blocks' button.
-		const isVisible = (
```

Comment:
It would be clearer to keep `isVisible` as a single expression because you can read one statement and understand completely what it means to be visible. Each subexpression can be commented if desired. Here, that knowledge is broken over three different control statements, so we have an increase in control flow and have to read multiple statements to follow what is happening and what it means to be visible.

In addition, each `if` statement encodes knowledge of what the return value looks like. If that shape changes, each return statement may need to be updated. If you'd like to keep the definition of `isVisible` broken up, perhaps it would be better assign `isVisible = false` in those cases and keep a single return statement.

Reply:
I've changed it to be a single expression again! 👍